pymeasure/instruments/keysight/keysight34980A.py

```diff
@@ -0,0 +1,520 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument, discreteTruncate, RangeException 
+import pyvisa as visa
+from pyvisa import VisaIOError
+
+class Keysight34980A(Instrument):
+    """ Represents the Keysight34980A Multifunction Switch/Measure Unit
+    and provides a high-level interface for taking scans with Keysight 34923A 
+    Multifunction Switch/Measure Unit.
+    """
+
+    def __init__(self,device_address):
+        print("Initializing Switch Matrix\n")
+        global rm 
+        rm = visa.ResourceManager()
+        self.switch_data = rm.open_resource(device_address)
+
+    #This command reads the instrument's (mainframe) identification string which contains 
+    #four comma-separated fields. The first field is the manufacturer's name, the second 
+    #field is the instrument model number, the third field is the serial number, and the 
+    #fourth field is a firmware revision code which contains four codes separated by dashes. 
+    def is_identification(self):
+        return(str(self.switch_data.query('*IDN?')))
+
+    #This command opens all channel relays and Analog Bus relays on the specified multiplexer 
+    #or switch module. Alternately by specifying the "ALL" parameter, this command opens 
+    #all channels on all installed modules. 
+    def set_channel_open_all(self, module = 1):    
+        self.switch_data.write(":ROUTe:OPEN:ALL " + str(module))
+
+    #This command opens the specified channel on a multiplexer or switch module.         
```

Comment:
This documentation string can better go into a docstring of one of the properties, or if a method cannot be converted into a property into the docstring format of that method, e.g.:
```
def some_method(self):
    """ Put the docstring over here
    Can also be multiline
    """

    # body of the method
```

This way, the documentation system that generates all the documentation knows where to find these docstrings and can use it for the doc-page of the instrument.